advanced_functionality/multi_model_linear_learner_home_value/linear_learner_multi_model_endpoint_inf_pipeline.ipynb

```diff
@@ -0,0 +1,1218 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Section 9 - Explore granular access to the target models of MME <a id='Finegrain-control-invoke-models'></a>\n",
+    "\n",
+    "If the role attached to this notebook instance allows invoking SageMaker endpoints, it is able to invoke all models hosted on the MME.  Using IAM conditional keys, you can restrict this model invocation access to specific models.  To explore this, you will create a new IAM role and IAM policy with conditional key to restrict access to a single model.  Assume this new role and verify that only a single target model can be invoked.\n",
+    "\n",
+    "Note that to execute this section, the role attached to the notebook instance should allow the following actions :\n",
+    "    \"iam:CreateRole\",\n",
+    "    \"iam:CreatePolicy\",\n",
+    "    \"iam:AttachRolePolicy\",\n",
+    "    \"iam:UpdateAssumeRolePolicy\"\n",
+    "    \n",
+    "If this is not the case, please work the Administrator of this AWS account to ensure this.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "iam_client = boto3.client('iam')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Create a new role that can be assumed by this notebook.  The roles should allow access to only a single model.\n",
+    "\n",
+    "path='/'\n",
+    "role_name='allow_invoke_ny_model_role'+strftime('%Y-%m-%d-%H-%M-%S', gmtime())\n",
+    "description='Role that allows invoking a single model'\n",
+    "\n",
+    "action_string = \"sts:AssumeRole\"\n",
+    "    \n",
+    "trust_policy={\n",
+    "  \"Version\": \"2012-10-17\",\n",
+    "  \"Statement\": [\n",
+    "    {\n",
+    "      \"Sid\": \"statement1\",\n",
+    "      \"Effect\": \"Allow\",\n",
+    "      \"Principal\": {\n",
+    "        \"AWS\": role\n",
+    "      },\n",
+    "      \"Action\": \"sts:AssumeRole\"\n",
+    "    }\n",
+    "  ]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response = iam_client.create_role(\n",
+    "    Path=path,\n",
+    "    RoleName=role_name,\n",
+    "    AssumeRolePolicyDocument=json.dumps(trust_policy),\n",
+    "    Description=description,\n",
+    "    MaxSessionDuration=3600\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role_arn=response['Role']['Arn']\n",
+    "print(\"Role arn is :\", role_arn)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint_resource_arn=\"arn:aws:sagemaker:\" + REGION + \":\" + ACCOUNT_ID + \":endpoint/\" + endpoint_name\n",
```

Comment:
Minor:
1. missing spaces around `=`.
1. the string can be formatted with `format()`.